derp/derphttp/derphttp_client.go

```diff
@@ -584,7 +584,9 @@ func (c *Client) dialNodeUsingProxy(ctx context.Context, n *tailcfg.DERPNode, pr
 		case <-done:
 			return
 		case <-ctx.Done():
-			proxyConn.Close()
+			if proxyConn != nil {
```

Comment:
how can this be nil after line 576?

Reply:
I am not sure, however this check seems to not be needed anymore so I removed it. It seems odd to me that it could consistently be nil.